design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,42 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol over three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as [vault](https://www.vaultproject.io/). Currently Rook configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW similar to `SSE-KMS`.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+
+## Proposal details
+Introducing new field `s3` in `SecuritySpec` which defines `AWS-SSE-S3` support for RGW.
+```
+security:
+  kms:
+    ..
+  s3:
+    connectionDetails:
+      KMS_PROVIDER: vault
+      VAULT_ADDR: https://vault.default.svc.cluster.local:8200
+      VAULT_SECRET_ENGINE: transit
+      VAULT_AUTH_METHOD: token
+    # name of the k8s secret containing the kms authentication token
+    tokenSecretName: rook-vault-token
+```
+The values for this configuration will be available in [Security field](Documentation/CRDs/Object-Storage/ceph-object-store-crd.md#security-settings) of `CephObjectStoreSpec`, so depending on the above option RGW can be configured with `SSE-KMS` and `SSE-S3` options.
```

Comment:
Please add a note that the options are not mutually exclusive and that they can be enabled/disabled independently. We should also note this in the doc update for users.